zmq.cpp

```diff
@@ -17,6 +17,19 @@ std::string z_recv(zmq::socket_t& socket){
     return std::string(static_cast<char*>(msg.data()), msg.size());
 }
 
+namespace {
+
+void purge_and_reply(zmq::socket_t& clients, int more, bool should_purge) {
```

Comment:
It is weird to have a `purge_and_reply` and sometimes don't actually purge...
You should maybe implement `purge`, then `send` (or not if it is a one line) and then `purge_and_reply` which will be a call to `purge` then `reply`. You won't need to deal with, `should_purge`